fiftyone/utils/plot/plotly.py

```diff
@@ -0,0 +1,1913 @@
+        customdata=ids,
+        mode="markers",
+        marker=marker,
+        hovertemplate=hovertemplate,
+    )
+
+    figure = pgo.Figure(scatter)
+
+    zoom, (center_lon, center_lat) = _compute_zoom_center(coords)
+    figure.update_layout(
+        mapbox_style="carto-positron",
+        mapbox=dict(center=dict(lat=center_lat, lon=center_lon), zoom=zoom),
+    )
+
+    return figure
+
+
+def _plot_scatter_mapbox_density(
+    coords,
+    values=None,
+    ids=None,
+    radius=None,
+    colorscale="Viridis",
+    value_label="value",
+):
+    hover_lines = []
+
+    if values is not None:
+        hover_lines = ["<b>%s: %%{}</b>" % value_label]
+
+    hover_lines.append("lat: %{lat:.5f}<br>lon: %{lon:.5f}")
+
+    if ids is not None:
+        hover_lines.append("ID: %{customdata}")
+
+    hovertemplate = "<br>".join(hover_lines) + "<extra></extra>"
+
+    density = pgo.Densitymapbox(
+        lat=coords[:, 1],
+        lon=coords[:, 0],
+        z=values,
+        radius=radius,
+        customdata=ids,
+        colorscale=colorscale,
+        hovertemplate=hovertemplate,
+    )
+
+    figure = pgo.Figure(density)
+
+    zoom, (center_lon, center_lat) = _compute_zoom_center(coords)
+    figure.update_layout(
+        mapbox_style="carto-positron",
+        mapbox=dict(center=dict(lat=center_lat, lon=center_lon), zoom=zoom),
+    )
+
+    return figure
+
+
+# source: https://stackoverflow.com/a/64148305
+def _compute_zoom_center(coords, pad=0.2, aspect_ratio=2.0):
+    min_lon, min_lat = coords.min(axis=0)
+    max_lon, max_lat = coords.max(axis=0)
+
+    center_lon = round(0.5 * (min_lon + max_lon), 5)
+    center_lat = round(0.5 * (min_lat + max_lat), 5)
+    center = (center_lon, center_lat)
+
+    # Longitudinal range by zoom level (20 to 1), in degrees, at equator
+    lon_zoom_range = np.array(
+        [
+            0.0007,
+            0.0014,
+            0.003,
+            0.006,
+            0.012,
+            0.024,
+            0.048,
+            0.096,
+            0.192,
+            0.3712,
+            0.768,
+            1.536,
+            3.072,
+            6.144,
+            11.8784,
+            23.7568,
+            47.5136,
+            98.304,
+            190.0544,
+            360.0,
+        ]
+    )
+
+    alpha = 1.0 + pad
+    height = (max_lat - min_lat) * alpha * aspect_ratio
+    width = (max_lon - min_lon) * alpha
+    lon_zoom = np.interp(width, lon_zoom_range, range(20, 0, -1))
+    lat_zoom = np.interp(height, lon_zoom_range, range(20, 0, -1))
+    zoom = int(round(min(lon_zoom, lat_zoom)))
+
+    return zoom, center
+
+
+def _patch_perform_plotly_relayout():
+    """Attempts to patch an unresolved issue with zooming/panning FigureWidgets
+    with Mapbox plots.
+
+    https://github.com/plotly/plotly.py/issues/2570
+    """
+    filepath = os.path.join(
+        os.path.dirname(os.__file__),
+        "site-packages",
+        "plotly",
+        "basedatatypes.py",
+    )
```

Comment:
This gives the incorrect path in my venv

Reply:
Hmm, can you patch it so it will either locate the file in other setups or just gracefully return?

Reply:
Will do